webpack/assets/javascripts/services/statisticsChartService.js

```diff
@@ -21,6 +22,37 @@ function getChartConfig(details) {
   return config;
 }
 
+function getChartConfigByData(_data, id, type) {
+  const filteredData = _data.filter(d => d[1] !== 0);
+  const data = filteredData.reduce(
+    (curr, next) => {
+      const key = next[0];
+      const names = {
+        ...curr.names,
+        [key]: key.replace(/_/g, ' ')
+      };
+
+      const colors = {
+        ...curr.colors,
+        [key]: next[2]
```

Comment:
in the server side, data is sent in the format [ [ key, value, color], [ key, value, color] ] . 
c3  accepts data in a uniqe way. colors and column names are passed by a dictionary, so reduce turns the array from the server into this specific format.
i've added a bit of comments to make things more readable hopefully...